test/simplify_algebra_test.cpp

```diff
@@ -2144,6 +2144,34 @@ TEST_CASE(simplify_split_add_relu_multi_axes)
     EXPECT(m1.sort() == m2.sort());
 }
 
+TEST_CASE(split_concat_extra_inputs)
+{
+    // Matched by matcher find_split_concat, but no substitution because the "slice"
+    // instructions in the input list have a different instruction mixed in
+    migraphx::module m1;
+    {
+        migraphx::shape s{migraphx::shape::float_type, {1, 3, 1, 1}};
+        migraphx::shape ws{migraphx::shape::float_type, {4, 3, 3, 3}};
+
+        auto x    = m1.add_parameter("x", s);
+        auto one  = m1.add_literal(migraphx::generate_literal(ws, 1));
+        auto conv = m1.add_instruction(migraphx::make_op("convolution"), one, x);
```

Comment:
We dont need to add a convolution, it also makes it harder to see what shape is being passed to slice/relu. Instead, just pass the `x` parameter to slice/relu directly and then adjust the shape of `x` parameter to match.

Reply:
If the convolution isn't there, it doesn't match.

Reply:
I removed the convolution and it does match:

```cpp
migraphx::shape s{migraphx::shape::float_type, {4, 1, 3, 3}};
auto x    = m1.add_parameter("x", s);
auto slice1 = m1.add_instruction(
    migraphx::make_op("slice", {{"axes", {2}}, {"starts", {0}}, {"ends", {1}}}), x);
auto relu   = m1.add_instruction(migraphx::make_op("relu"), x);
auto slice2 = m1.add_instruction(
    migraphx::make_op("slice", {{"axes", {2}}, {"starts", {1}}, {"ends", {3}}}), x);
auto concat =
    m1.add_instruction(migraphx::make_op("concat", {{"axis", 2}}), slice1, relu, slice2);

m1.add_instruction(pass_op{}, concat);
```

And if I remove the extra check you added, I also get `std::bad_cast` error from this test, so this covers the case without needing the convolution.

Reply:
You are right.  Updated